swarm/src/behaviour.rs

```diff
@@ -488,6 +494,7 @@ pub struct ListenFailure<'a> {
     pub send_back_addr: &'a Multiaddr,
     pub error: &'a ListenError,
     pub connection_id: ConnectionId,
+    pub peer_id: Option<PeerId>,
```

Comment:
I think this might be a useful addition in itself, however, I don't think it is needed for this particular case.

Mind posting this up as a separate PR?